### ISSUE TITLE
Checkbox in a button group on Mobile Safari.

### ISSUE THREAD

**Issue by gubaidulin** [twbs/bootstrap#16812]
When a checkbox is put in 'btn-group', on iOS 8 it 'breaks' a button group.  

Code: https://jsfiddle.net/a4usrt4s/

![bug](https://cloud.githubusercontent.com/assets/3873027/8638390/aa0a2112-28c2-11e5-9911-a6da4a9c4a80.png)

**Comment by mdo** [comment]
What reason would you have for putting a tappable input in a tappable button?

**Comment by NogsMPLS** [comment]
buttongroups with checkboxes is documented here: http://getbootstrap.com/javascript/#buttons-checkbox-radio

as @mdo said, by exposing the actual checkbox visually inside a button, you are conflating two input concepts at once to a viewer.

The purpose of a 'checkbox in a button' is to give the user the ability to select multiple _descriptive_ buttons/checkboxes inside a form to submit or interact with. I say descriptive because a checkbox on its own -- inside a button no less -- has no meaning. If you are using a checkbox inside a 2 button button group to denote a kind of 'check this button to use this next button', then perhaps you should merely get rid of the checkbox and use the singular button with some descriptive text.

All that said though, there are always weird use cases to think of. With that in mind, perhaps it is better to do the best of both worlds and use a glyphicon or some similar image/css combo? that way different browsers rendering different sized checkboxes will not be an issue. Something like this maybe?

https://jsfiddle.net/a4usrt4s/8/

Though admittedly, it seems this has some styling issues as well. Looks like the ```top: 1px``` on the glyphicon class messes it up a little bit. Could be easily enough fixed with something like ```.btn.glyphicon {top: 0px;}``` or something like that. Should work across all browsers and gives you flexibility to use whatever styling you wish.

**Comment by twbs-lmvtfy** [comment]
Hi @NogsMPLS!

You appear to have posted a live example (https://fiddle.jshell.net/a4usrt4s/8/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 50, column 3: [E031](https://github.com/twbs/bootlint/wiki/E031): Glyphicon classes must only be used on elements that contain no text content and have no child elements.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by NogsMPLS** [comment]
hmmm...well, i suppose you are right Mr. Lint-Bot...but i still think its a good usage of glyphicon for this case.

either way, i did accidentally link the wrong jsfiddle, tho i suppose the bot will not like this one either.

https://jsfiddle.net/a4usrt4s/9/

**Comment by twbs-lmvtfy** [comment]
Hi @NogsMPLS!

You appear to have posted a live example (https://fiddle.jshell.net/a4usrt4s/9/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 50, column 3: [E031](https://github.com/twbs/bootlint/wiki/E031): Glyphicon classes must only be used on elements that contain no text content and have no child elements.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by NogsMPLS** [comment]
okay, fine, this should work then.

https://jsfiddle.net/a4usrt4s/10/

**Comment by twbs-lmvtfy** [comment]
Hi @NogsMPLS!

You appear to have posted a live example (https://fiddle.jshell.net/a4usrt4s/10/show/light/), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 51, column 49: [E031](https://github.com/twbs/bootlint/wiki/E031): Glyphicon classes must only be used on elements that contain no text content and have no child elements.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by NogsMPLS** [comment]
hrm, whitespace, picky robot.

https://jsfiddle.net/a4usrt4s/11/

**Comment by gubaidulin** [comment]
@mdo, Gmail has the same element for selecting all mails.

**Comment by cvrebert** [comment]
@mdo So, what's the opinion on this?

**Comment by mdo** [comment]
Not much of a way around it I feel—resizing iOS form controls sounds like a shitshow to me right now :). Closing as a won't fix, though some focused CSS overrides and browser sniffing might get you close.
